### PR TITLE
Fix WalletConnect stuck forever bug

### PR DESCRIPTION
**Pull request by han0110**
## Problem

1. When user close the QRCode modal, developer has to reset the `walletConnectProvider` to make next calling `enable` to work, like [Uniswap](https://github.com/Uniswap/uniswap-interface/blob/8975086a691ccac35fc76af21e3b2e6f39469fe5/src/components/WalletModal/index.tsx#L183-L186)'s workaround. It is reproducible and well-described in #124.
2. When user scan the QRCode, however, reject the connection, the modal will just disappear without any error for us to handle.

## Workaround

1. Reset `walletConnectProvider` when user close the modal.
2. Add event callback for `disconnect` in `walletConnectProvider.wc` to bubble up the error when user reject the connection.

## Others

1. Upgrade `@walletconnect/provider` to `1.3.4` and use type in `@walletconnect/types` for constructor option
2. Support multiple `rpc` url with `chainId` configurable in constructor option (reference [here](https://github.com/lootex-io/web3-react/blob/2b10de90ef2727f9288365b9f224d91d19a12992/example/connectors.ts#L28-L34))
3. Add `.gitignore` to ignore `.next` when developing with example

### CODE REVIEW

**Comment by jotefer** [comment]
any news about implementing this fix?

**Comment by han0110** [comment]
Update:
1. Pass CI
2. Upgrade `@walletconnect/provider`  to  `1.3.4`

File: packages/walletconnect-connector/src/index.ts [hunk 1]
```diff
@@ -88,18 +89,36 @@ export class WalletConnectConnector extends AbstractConnector {
...
+    let account: string
+    account = await new Promise<string>((resolve, reject) => {
+      const userReject = () => {
+        // Erase the provider manually
+        this.walletConnectProvider = undefined
+        reject(new UserRejectedRequestError())
+      }
+
+      // Workaround to bubble up the error when user reject the connection
+      this.walletConnectProvider.wc.on('disconnect', (_: Error, event: IInternalEvent) => {
```
Comment (self-coding-crab) [inline packages/walletconnect-connector/src/index.ts]: On my end, I'm unable to detect 'disconnect' action.
Is this action dispatched when we click 'close' button on trustwallet ( to reject session ) after scanning QR code?

**Comment by self-coding-crab** [comment]
Could ayone merge this pr asap?

File: packages/walletconnect-connector/src/index.ts (hunk 1 above)
Comment (han0110) [reply packages/walletconnect-connector/src/index.ts]: Yes, the `disconnect` event emitted when user click close button to reject the session connection. I have tested and worked for me on these mobile wallet:
* Metamask
* Trust Wallet
* Argent

By the way I just follow the official [README](https://github.com/WalletConnect/walletconnect-monorepo/blob/1.3.4/packages/clients/client/README.md#initiate-connection) and not so sure whether it works for all mobile wallet. Actually the package `@walletconnect/web3-provider` this one depending on also [implements similar logic](https://github.com/WalletConnect/walletconnect-monorepo/blob/b2cfd26b5648cf0fb884b8115a288977701d5116/packages/providers/web3-provider/src/index.ts#L345-L351) for this situation but don't know why it doesn't work.

**Comment by jackblackCH** [comment]
Any news here?

**Comment by NoahZinsmeister** [comment]
hey all - sorry for the delayed response. appreciate the efforts here, but going forward i'm focusing on the v8 version of this library, which has a walletconnect connector! happy to accept PRs against this in the `main` branch

**Comment by NoahZinsmeister** [comment]
closing in favor of #360

**Pull request closed without merging**